### PR TITLE
DRILL-8118: Add Option to Allow Disk Use on Mongo Queries

### PR DESCRIPTION
**Pull request by cgivre**
# [DRILL-8118](https://issues.apache.org/jira/browse/DRILL-8118): Add Option to Allow Disk Use on Mongo Queries

## Description
MongoDB has a strange feature  whereby queries which use more than 100MB of memory will by default fail.  Mongo allows the user to specify whether they want the query to spill to disk which allows larger queries but at a performance cost.
This minor PR adds the ability for a user to specify whether they want this option included in Mongo queries.  This only affects aggregate queries in Mongo. 

## Documentation
This adds a new config option to the mongo config: `allowDiskUse` which, when set to `true`, will pass that parameter down to MongoDB during aggregate queries.  

## Testing
Ran unit tests

### CODE REVIEW

File: contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/MongoRecordReader.java [hunk 1]
```diff
@@ -198,7 +198,11 @@ public int next() {
         if (!fields.isEmpty()) {
           operations.add(Aggregates.project(fields));
         }
-        projection = collection.aggregate(operations);
+        if (plugin.getConfig().allowDiskUse()) {
+          projection  = collection.aggregate(operations).allowDiskUse(true);
```
Comment (luocooong) [inline contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/MongoRecordReader.java:202]:
```suggestion
          projection = collection.aggregate(operations).allowDiskUse(true);
```

File: contrib/storage-mongo/src/main/resources/bootstrap-storage-plugins.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "mongo" : {
       "type":"mongo",
       "connection":"mongodb://localhost:27017/",
+      "allowDiskUse": false,
```
Comment (luocooong) [inline contrib/storage-mongo/src/main/resources/bootstrap-storage-plugins.json:6]: Is it possible that the old stored plugin objects were not able to deserialize from the ZK (based on the distributed ZK)?

**Review by luocooong — COMMENTED** [review]
@cgivre Thanks for the contribution, I have a very small comment.

File: contrib/storage-mongo/src/main/resources/bootstrap-storage-plugins.json (hunk 2 above)
Comment (cgivre) [reply contrib/storage-mongo/src/main/resources/bootstrap-storage-plugins.json:6]: @luocooong, Thanks for the quick review.  The deserialization should only be affected if you remove config parameters. This defaults to `false`, so if the user doesn't do anything, Drill's behavior stays the same.

File: contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/MongoRecordReader.java (hunk 1 above)
Comment (cgivre) [reply contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/MongoRecordReader.java]: Fixed!

**Review by luocooong — APPROVED** [review]
LGTM. +1

**Pull request merged**